docs/usage/type-checking/index.md

```diff
@@ -37,12 +37,12 @@ const options = {
 };
 
 const program = Linter.createProgram("tsconfig.json", "projectDir/");
+const linter = new Linter(options, program);
 
 const files = Linter.getFileNames(program);
 const results = files.map(file => {
     const fileContents = program.getSourceFile(file).getFullText();
     const configuration = Configuration.findConfiguration(configurationFilename, file).results;
-    const linter = new Linter(options, program);
     linter.lint(file, fileContents, configuration);
     return linter.getResult();
```

Comment:
this will include the failures of the current file and all previously linted files.
It may be better to move this one out of the loop. You could then replace the `.map` with `.forEach` or a simple `for..of` loop.

Reply:
Oh! Didn't know that. Based on the old example I always thought it would return the "last" result or something like that. I'll change that :)

Reply:
Crazy. I _never_ noticed this, because the old example also used `map(... => linter.getResult())`.